src/app/modules/examples/__tests__/examples.controller.spec.ts

```diff
@@ -0,0 +1,208 @@
+import { errAsync, okAsync } from 'neverthrow'
+import { mocked } from 'ts-jest/utils'
+
+import { Colors } from 'src/types'
+
+import expressHandler from 'tests/unit/backend/helpers/jest-express'
+
+import { DatabaseError } from '../../core/core.errors'
+import * as ExamplesController from '../examples.controller'
+import { ResultsNotFoundError } from '../examples.errors'
+import { ExamplesFactory } from '../examples.factory'
+import { SingleFormResult } from '../examples.types'
+
+jest.mock('../examples.factory')
+const MockExamplesFactory = mocked(ExamplesFactory)
+
+describe('examples.controller', () => {
+  beforeEach(() => jest.clearAllMocks())
+
+  describe('handleGetExamples', () => {
+    const MOCK_REQ_QUERY = {
+      pageNo: '1',
+    }
+    const MOCK_REQ = expressHandler.mockRequest({
+      query: MOCK_REQ_QUERY,
+      session: {
+        user: 'exists',
+      },
+    })
+
+    it('should return 200 with an array of example forms', async () => {
+      // Arrange
+      const mockRes = expressHandler.mockResponse()
+      // Mock getExampleForms to return error.
+      const mockResult = {
+        forms: [],
+        totalNumResults: 0,
+      }
+      MockExamplesFactory.getExampleForms.mockReturnValueOnce(
+        okAsync(mockResult),
+      )
+
+      // Act
+      await ExamplesController.handleGetExamples(MOCK_REQ, mockRes, jest.fn())
+
+      // Assert
+      expect(MockExamplesFactory.getExampleForms).toHaveBeenCalledWith(
+        MOCK_REQ_QUERY,
+      )
+      expect(mockRes.status).toBeCalledWith(200)
+      expect(mockRes.json).toBeCalledWith(mockResult)
+    })
+
+    it('should return 401 when user is not in session', async () => {
+      // Arrange
+      const mockReqNoSession = expressHandler.mockRequest({
+        query: MOCK_REQ_QUERY,
```

Comment:
hm, this makes me wonder whether our check for the user being in session is too lax. should we be checking that the user object has a certain shape, instead of just checking for truthiness?

Reply:
Maybe I'll check for `user.id`

Reply:
Will do in a separate PR to avoid polluting this PR even further